mypy/typeanal.py

```diff
@@ -784,7 +785,13 @@ def visit_raw_expression_type(self, t: RawExpressionType) -> Type:
         # make signatures like "foo(x: 20) -> None" legal, we can change
         # this method so it generates and returns an actual LiteralType
         # instead.
-
+        # Sounds based to me
+        if (
+            self.api.is_future_flag_set("annotations")
+            and t.base_type_name in ('builtins.int', 'builtins.bool')
```

Comment:
support `str` and just ban quoted types when `__future__.annotations` is set

Reply:
that sounds like more of a disruptive change, maybe in a follow-up

Reply:
#361